substrate/primitives/genesis-builder/src/lib.rs

```diff
@@ -49,6 +57,22 @@ sp_api::decl_runtime_apis! {
 		/// It is recommended to log any errors encountered during the process.
 		///
 		/// Please note that provided json blob must contain all `GenesisConfig` fields, no defaults will be used.
-		fn build_config(json: sp_std::vec::Vec<u8>) -> Result;
+		#[renamed("build_config", 2)]
+		fn build_state(json: sp_std::vec::Vec<u8>) -> Result;
+
+		/// Returns a JSON blob representation of the builtin `GenesisConfig` identified by `id`.
+		///
+		/// If `id` is `None` the function instantiates the default `GenesisConfig` struct for the runtime and serializes it into a JSON
+		/// blob. It returns a `Vec<u8>` containing the JSON representation of the default `GenesisConfig`.
+		/// Otherwise returns a JSON representation of the built-in, named, pre-configured `GenesisConfig` value.
+		#[api_version(2)]
+		fn get_preset(id: Option<sp_std::vec::Vec<u8>>) -> Option<sp_std::vec::Vec<u8>>;
```

Comment:
maybe:
```suggestion
		fn get_preset(id: Option<RuntimeString>) -> Option<sp_std::vec::Vec<u8>>;
```

Reply:
I had some troubles with using RuntimeString in [match](https://github.com/paritytech/polkadot-sdk/blob/249b8136ad19addd97e73e01f426e990d543718a/substrate/test-utils/runtime/src/lib.rs#L758-L760) , don't really see much benefit from having it here.

Reply:
`String` :facepalm:

Reply:
If the `preset_names` return Strings and the `get_preset` accepts `Vec<u8>` then it is up to the API user to figure out that he is supposed to UTF8 encode that?  
Maybe use `Vec<u8>` or `RuntimeString` everywhere?